components/supervisor/pkg/terminal/terminal.go

```diff
@@ -146,35 +150,49 @@ func (m *Mux) doClose(alias string, gracePeriod time.Duration) error {
 	return nil
 }
 
-func (term *Term) gracefullyShutdownProcess(gracePeriod time.Duration) error {
+func (term *Term) gracefullyShutdownProcess(sig syscall.Signal) (err error) {
 	if term.Command.Process == nil {
-		// process is alrady gone
+		// process is already gone
 		return nil
 	}
-	if gracePeriod == 0 {
+
+	if sig == unix.SIGKILL {
```

Comment:
we should either change the method (remove 'gracefully') name or not allow SIGKILL her but handle it on the call side.

Reply:
thinking about it, I think we should always send SIGHUP followed by SIGINT if the process is still alive. So the method name would be good, but we should not pass the signal?